src/app/large-display/large-display.component.ts

```diff
@@ -111,31 +151,82 @@ export class LargeDisplayComponent implements OnInit, AfterViewInit {
     }
   ];
 
-  running: boolean;
+  currentScreen: ScreenId = ScreenId.WaitingToAuthenticate;
+  running: boolean = null;
 
-  constructor(public timerService: TimerService) { }
+  constructor(
+    public timerService: TimerService,
+    public authService: AuthService,
+    public firebaseService: FirebaseService,
+    private zone: NgZone,
+  ) { }
 
   // TODO: These values are only here for testing. Eventually, we'll get this
   // information from the round service.
   public startTime = TimePeriod.fromMonthAndQuarter(4, 1);
   public endTime = TimePeriod.fromMonthAndQuarter(11, 4);
 
+  // TODO: For the moment, we're only using one fixed, preexisting round for
+  // all teachers. Eventually, teachers will each create their own sessions
+  // and rounds.
+  readonly roundPath = {sessionId: 'demo-session', roundId: 'demo-round'};
+
   // The flowers displayed are essentially the demoFlowers at this moment
+
   ngOnInit() {
+    this.authService.logTeacherIn().then(() => {
+      // At time of this writing, AngularFire uses a DIY implementation of the
+      // Promise class that runs outside of the Angular Zone. As a result, code
+      // that runs inside this callback won't trigger change detection unless
+      // we tell it to.
+      // See https://github.com/google/google-api-javascript-client/issues/353
+      this.zone.run(() => {
+        this.currentScreen = ScreenId.WaitingToStartTheRound;
+      });
+    });
+  }
+
+  /**
+   * Create a new round within the session, set its initial state, and make .
+   *
+   * TODO: As of this iteration, this function just re-uses the old round and
+   * re-populates it with data.
+   */
+  startRound() {
+    this.currentScreen = ScreenId.DuringTheRound;
+
+    // Initialize the round in Firestore.
+    // TODO: Eventually, we'll create a whole new round, not just update an
+    // old one.
+    this.firebaseService.updateRoundData(this.roundPath, {
+      running: false,
+      currentTime: this.startTime.time,
+      flowerSpeciesIds: this.demoFlowerSpecies.map(species => species.id),
+    });
+
+    // Give the timer its starting state.
     this.timerService.initialize({
       running: false,
       tickSpeed: 1000,
       currentTime: this.startTime,
       endTime: this.endTime
     });
 
+    // Keep a copy of the "running" boolean for the large display controls to
+    // use.
     this.timerService.running$.subscribe(running => {
       this.running = running;
     });
```

Comment:
I feel like if things in the large display can use the `async` pipe, they should. If we need subscriptions like this we should make sure they get unsubscribed properly

Reply:
Done, thanks!